cloud/new_ec2.go

```diff
@@ -139,19 +139,21 @@ func (m *ec2Manager) spawnOnDemandHost(h *host.Host, ec2Settings *NewEC2Provider
 	}
 
 	grip.Debug(message.Fields{
-		"message":       "starting on-demand instance",
-		"args":          input,
-		"host":          h.Id,
-		"host_provider": h.Distro.Provider,
-		"distro":        h.Distro.Id,
+		"cloud_provider": "ec2",
+		"message":        "starting on-demand instance",
+		"args":           input,
+		"host":           h.Id,
+		"host_provider":  h.Distro.Provider,
+		"distro":         h.Distro.Id,
 	})
 	reservation, err := m.client.RunInstances(input)
 	if err != nil || reservation == nil {
 		grip.Error(message.WrapError(h.Remove(), message.Fields{
-			"message":       "error removing intent host",
-			"host":          h.Id,
-			"host_provider": h.Distro.Provider,
-			"distro":        h.Distro.Id,
+			"cloud_provider": "ec2",
+			"message":        "error removing intent host",
+			"host":           h.Id,
+			"host_provider":  h.Distro.Provider,
+			"distro":         h.Distro.Id,
```

Comment:
we don't get the reason why we're removing the intent host here, (either err, or reservation,) so this is only logged if we have trouble with the database connection.

also cloud provider and host provider are probably always mostly the same thing? or at least cloud provider won't add much information.

Reply:
does h.Remove() set h to nil?

Reply:
> we don't get the reason why we're removing the intent host here, (either err, or reservation,) so this is only logged if we have trouble with the database connection.

Fixed.

> also cloud provider and host provider are probably always mostly the same thing? or at least cloud provider won't add much information.

Host is not always available where I use `cloud_provider`, like in the client wrappers. By adding this field, we can search Splunk for everything this provider is up to, which is useful. I made it a constant though, which it should have been.

> does h.Remove() set h to nil?

It removes it from the collection, but it does not set h to nil.